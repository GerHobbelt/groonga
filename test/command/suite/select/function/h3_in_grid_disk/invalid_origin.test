#@on-error omit
plugin_register functions/h3
#@on-error default

table_create Data TABLE_HASH_KEY ShortText
column_create Data location COLUMN_SCALAR WGS84GeoPoint

load --table Data
[
{"_key": "ClearCode: 35.80460x139.45649", "location": "35.80460x139.45649"},
{"_key": "Shin Tokorozawa: 35.80747x139.45602", "location": "35.80747x139.45602"},
{"_key": "Koku koen: 35.79841x139.46573", "location": "35.79841x139.46573"}
]

select Data --filter 'h3_in_grid_disk(location, 29, 10, 3)'
