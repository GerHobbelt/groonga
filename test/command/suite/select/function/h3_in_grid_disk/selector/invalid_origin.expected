plugin_register tokenizers/h3_index
[[0,0.0,0.0],true]
plugin_register functions/h3
[[0,0.0,0.0],true]
table_create H3Indexes TABLE_PAT_KEY UInt64   --default_tokenizer 'TokenH3Index("resolution", 10)'
[[0,0.0,0.0],true]
table_create Data TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
column_create Data location COLUMN_SCALAR WGS84GeoPoint
[[0,0.0,0.0],true]
column_create H3Indexes data_location COLUMN_INDEX Data location
[[0,0.0,0.0],true]
load --table Data
[
{"_key": "ClearCode: 35.80460x139.45649", "location": "35.80460x139.45649"},
{"_key": "Shin Tokorozawa: 35.80747x139.45602", "location": "35.80747x139.45602"},
{"_key": "Koku koen: 35.79841x139.46573", "location": "35.79841x139.46573"}
]
[[0,0.0,0.0],3]
select Data --filter 'h3_in_grid_disk(location, 29, 10, 3)'
[[[-22,0.0,0.0],"h3_in_grid_disk(): origin must be WGS84GeoPoint: 29"]]
#|e| h3_in_grid_disk(): origin must be WGS84GeoPoint: 29
